File: features/snippets/external_storage/s3_setup/s3_external_storage_setup.py
import dataclasses

from temporalio.client import Client, ClientConfig
from temporalio.converter import DataConverter, ExternalStorage
from temporalio.worker import Worker


async def setup(driver):
    # @@@SNIPSTART python-s3-external-storage-setup
    data_converter = dataclasses.replace(
        DataConverter.default,
        external_storage=ExternalStorage(drivers=[driver]),
    )

    client_config = ClientConfig.load_client_connect_config()

    client = await Client.connect(**client_config, data_converter=data_converter)

    worker = Worker(
        client,
        task_queue="my-task-queue",
        workflows=[],
        activities=[],
    )
    # @@@SNIPEND
    await worker.run()

File: features/snippets/external_storage/s3_setup/s3_external_storage_setup.ts
/* eslint-disable @typescript-eslint/no-unused-vars */
// @@@SNIPSTART typescript-s3-external-storage-setup
import { Client, Connection } from '@temporalio/client';
import { ExternalStorage } from '@temporalio/common';
import { Worker } from '@temporalio/worker';

async function externalStorageSetup() {
  const dataConverter = {
    externalStorage: new ExternalStorage({ drivers: [driver] }),
  };

  const connection = await Connection.connect();
  const client = new Client({ connection, dataConverter });

  const worker = await Worker.create({
    workflowsPath: require.resolve('./workflows'),
    taskQueue: 'my-task-queue',
    dataConverter,
  });
}
// @@@SNIPEND

declare const driver: import('@temporalio/common').StorageDriver;
